x-pack/test/functional_with_es_ssl/apps/triggers_actions_ui/details.ts

```diff
@@ -331,5 +331,94 @@ export default ({ getPageObjects, getService }: FtrProviderContext) => {
         await pageObjects.alertDetailsUI.ensureAlertInstanceExistance('eu-east', false);
       });
     });
+
+    describe('Alert Instance Pagination', function() {
+      const testRunUuid = uuid.v4();
+      let alert: any;
+
+      before(async () => {
+        await pageObjects.common.navigateToApp('triggersActions');
+
+        const actions = await Promise.all([
+          alerting.actions.createAction({
+            name: `server-log-${testRunUuid}-${0}`,
+            actionTypeId: '.server-log',
+            config: {},
+            secrets: {},
+          }),
+          alerting.actions.createAction({
+            name: `server-log-${testRunUuid}-${1}`,
+            actionTypeId: '.server-log',
+            config: {},
+            secrets: {},
+          }),
+        ]);
+
+        const instances = flatten(
+          range(10).map(index => [
+            { id: `us-central-${index}` },
+            { id: `us-east-${index}` },
+            { id: `us-west-${index}` },
+          ])
+        );
+        alert = await alerting.alerts.createAlwaysFiringWithActions(
+          `test-alert-${testRunUuid}`,
+          actions.map(action => ({
+            id: action.id,
+            group: 'default',
+            params: {
+              message: 'from alert 1s',
+              level: 'warn',
+            },
+          })),
+          {
+            instances,
+          }
+        );
+
+        // refresh to see alert
+        await browser.refresh();
+
+        await pageObjects.header.waitUntilLoadingHasFinished();
+
+        // Verify content
+        await testSubjects.existOrFail('alertsList');
+
+        // click on first alert
+        await pageObjects.triggersActionsUI.clickOnAlertInAlertsList(alert.name);
+
+        // await first run to complete so we have an initial state
+        await retry.try(async () => {
+          const { alertInstances } = await alerting.alerts.getAlertState(alert.id);
+          expect(Object.keys(alertInstances).length).to.eql(instances.length);
+        });
+      });
+
+      it('renders the first page', async () => {
+        // Verify content
+        await testSubjects.existOrFail('alertInstancesList');
+
+        const { alertInstances } = await alerting.alerts.getAlertState(alert.id);
+
+        const [firstItem] = await pageObjects.alertDetailsUI.getAlertInstancesList();
+        expect(firstItem.instance).to.eql(Object.keys(alertInstances)[0]);
```

Comment:
I'm curious if the reason this works is because the instances are returned in a sorted order by `getAlertState()` - or somehow ordered by time.  Seems like it could be flaky if there's some async, or extracting data from ES that ends up not being sorted, or something.

Reply:
Yup, this will likely have to change once we're using the event log.
For now it works because the order of keys returned by Object.keys is the same in both places, but as we don't really have a clear order for Alert Instances, this seems unavoidable.
Even ordering by start will have the same randomness as they all, by definition, have the same value for start. 😆 

We _could_ order by the ID, but that's what Object.values already does. I feel like as this is temporary until we have the Event Log API done, this is good enough.

Reply:
Ya, makes sense.  Seems like we will have to come up with some kind of ordering, just to be consistent, so once we get there, this test will have to changed to reflect that.